keps/487-kubectl-plugin-listing/README.md

```diff
@@ -0,0 +1,316 @@
+The code will live in cmd/ directory.
+
+The CLI command will have the following format:
+
+```bash
+$kubectl kueue <type> <command> <name> <flags> <options>
+```
+
+### Version
+
+```bash
+$kubectl kueue version
+kueue pod was not found in kueue-system namespace
+
+$kubectl kueue version
+kueue:v0.5.0
+```
+
+`kubectl kueue version` command will verify that kueue is running by checking the `kueue-controller-manager` pod in kueue-system namespace and return the version of kueue based on the image name in pod spec.
+
+### Describe
+
+```bash
+$kubectl kueue workloads describe job-team-a --namespace=team-a --status
+State: Pending
+Status
+  Conditions:
+    Last Transition Time:  2023-11-09T09:26:58Z
+    Message:               couldn't assign flavors to pod set main: resource ephemeral-storage unavailable in ClusterQueue
+    Reason:                Pending
+    Status:                False
+    Type:                  QuotaReserved
+
+$kubectl kueue workload describe job-team-a --namespace team-a --spec
+Spec:
+  Pod Sets:
+    Count:  3
+    Name:   main
+    Template:
+      Metadata:
+        Labels:
+          batch.kubernetes.io/controller-uid:  37a03eb0-5c59-41f2-be3c-e9e5ca75bc7c
+          batch.kubernetes.io/job-name:        sample-job-team-a-vdjg9
+          Controller - UID:                    37a03eb0-5c59-41f2-be3c-e9e5ca75bc7c
+          Job - Name:                          sample-job-team-a-vdjg9
+      Spec:
+        Containers:
+          Args:
+            10s
+          Image:              gcr.io/k8s-staging-perf-tests/sleep:latest
+          Image Pull Policy:  Always
+          Name:               dummy-job
+          Resources:
+            Limits:
+              Cpu:                  500m
+              Ephemeral - Storage:  512Mi
+              Memory:               512Mi
+              nvidia.com/gpu:       1
+            Requests:
+              Cpu:                     500m
+              Ephemeral - Storage:     512Mi
+              Memory:                  512Mi
+              nvidia.com/gpu:          1
+          Termination Message Path:    /dev/termination-log
+          Termination Message Policy:  File
+        Dns Policy:                    ClusterFirst
+        Node Selector:
+          cloud.google.com/gke-accelerator:  nvidia-tesla-t4
+        Restart Policy:                      Never
+        Scheduler Name:                      default-scheduler
+        Security Context:
+        Termination Grace Period Seconds:  30
+  Priority:                                0
+  Priority Class Source:                   
+  Queue Name:                              lq-team-a
+
+```
+
+In order to make kueue CLI simple, we are going to have only one abstraction for jobs specific information — workload. Currently the `describe` command will mostly duplicate the `kubectl describe` command with few modifications:
+`--status` option will display Workload.Status field and state of Workload.
```

Comment:
Why do we need these --status --spec options?

Reply:
User wants to know why the workload is pending or was there any errors during admission. I suggest to add --status option that will limit describe command output and will provide information for the users needs.
For --spec option I'm not so confident if this is needed. My thoughts were is that user would like to verify if the spec is correct when they've seen some error in admission check.